Investigations Print Preview Page

Jacobjeevan opened the pull request:
## Proposed Changes

- Fixes #8583 
- Created a different print preview page for investigations, similar to prescriptions print preview (Medications tab)
- Modified Investigations view page to have no scrollbar (+ print button redirects to print preview page instead)

### Investigations tab

<img width="1226" alt="image" src="https://github.com/user-attachments/assets/a1d45b7a-4702-402d-8d14-e9b4bd19f61f">

### Investigations View Page

<img width="1010" alt="image" src="https://github.com/user-attachments/assets/40c228b5-aa60-4eda-b4a2-f57d7907e255">

<img width="504" alt="image" src="https://github.com/user-attachments/assets/a892dec6-8d08-4583-8cc8-eb71f7c3e9df">


### Investigations Print Preview Page

<img width="943" alt="image" src="https://github.com/user-attachments/assets/37dfdcb9-8eb9-4f7c-adf7-57ef2d11710b">


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA

Jacobjeevan committed:
Investigations View

- Created a different print preview page for investigations, similar to medications
- Changed the existing Investigations print page to just view page (i.e. print button redirects to print preview page instead)

rithviknishad commented on src/Components/Facility/Investigations/ViewInvestigations.tsx:
```diff
@@ -58,16 +60,31 @@ export default function ViewInvestigations(props: {
+                    ghost
+                    border
+                  >
+                    View
```
can we keep this in i18n

rithviknishad commented on src/Components/Facility/Investigations/ViewInvestigations.tsx:
```diff
@@ -58,16 +60,31 @@ export default function ViewInvestigations(props: {
+                    border
+                  >
+                    <CareIcon icon="l-print" className="text-lg" />
+                    Print
```
i18n

rithviknishad commented on src/Components/Facility/Investigations/InvestigationTable.tsx:
```diff
@@ -127,17 +127,15 @@ export const InvestigationTable = ({
-                    </th>
-                  ),
-                )}
+                {["Name", "Value", "Ideal"].map((heading) => (
```
i18n

rithviknishad reviewed (commented):
LGTM, just i18n

Jacobjeevan committed:
Investigations

- Disable navigate on save for investigations form edit
	- Also replace the initial values after submit (correct behavior when clicking cancel following a form submission)
- Using translations for terms
- Adjusted width of table in view page

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

Jacobjeevan committed:
added comma

Jacobjeevan commented:
>   * [x]  in mobile view, the inputted data in the field is not visible
>   * [x]  reduce the width of the value field in normal view, as its not that much required
>   * [x]  when we save the edited value, the users have been redirected to the consultation dashboard, it shouldn't be the expected behavior, keep the user in the same page itself to view the updated value

👍  @nihal467

nihal467 commented:
![image](https://github.com/user-attachments/assets/961f3f02-48af-41c6-b12f-9f85600293e8)

@aparnacoronasafe can you look into the new investigation print preview, without requesting @Jacobjeevan added patient admission, IP/OP no, allergies, and bed no to the print preview of the investigation, is it ok to keep it or should we remove them and only keep the patient name and facility name in the investigation print preview

bodhish commented:
Lets add it; the prints should eventually have everything we store.

Jacobjeevan committed:
Removed patient details from Investigations print page

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

Jacobjeevan commented:
@nihal467 I was working on #8275, should we do the same thing here and move print button inside?

<img width="967" alt="image" src="https://github.com/user-attachments/assets/9daf362b-3cb8-4066-874c-dda812c21ede">


<img width="970" alt="image" src="https://github.com/user-attachments/assets/e85147ce-bd80-4c95-ab57-5cad18dead1d">


<img width="365" alt="image" src="https://github.com/user-attachments/assets/3c61c2c8-8cc5-429a-877a-541f347fc9f9">

nihal467 commented:
@Jacobjeevan that would be good to do it

Jacobjeevan committed:
Moved print button inside body for InvestigationTable

Jacobjeevan commented:
> @Jacobjeevan yes we can do it and once the PR is ready, let me know

Ready 👍

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

nihal467 commented:
![image](https://github.com/user-attachments/assets/4a5fc76b-e37a-4dc9-8295-1923b6f40b42)

@Jacobjeevan the mobile responsive bug is back, in mobile, if you type three-digit, you are not able to view it together

Jacobjeevan committed:
adjusting the widths for table

Jacobjeevan committed:
fix widths

Jacobjeevan commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/374617735-4a5fc76b-e37a-4dc9-8295-1923b6f40b42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NDY4NDksIm5iZiI6MTcyODQ0NjU0OSwicGF0aCI6Ii81NzA1NTk5OC8zNzQ2MTc3MzUtNGE1ZmM3NmItZTM3YS00ZGM5LTgyOTUtMTkyM2I2ZjQwYjQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA5VDA0MDIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjY2RiNzNkZTYwMDc2ZjE2NGU0YWVhYWU5MTUxNmI1N2VmZjRmZjI5ODUyNjFjZjIzMGM5ZTcwNzBmMjY5M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0meT7kOLcw2bJbITxg1ySo6PQKNG_ZViR7FWGg47Vw4)
> 
> @Jacobjeevan the mobile responsive bug is back, in mobile, if you type three-digit, you are not able to view it together

Weird, I checked firefox and safari and it's fine for me 🤔 . But, I've adjusted the widths, take a look and see if the issue persists. 

<img width="571" alt="image" src="https://github.com/user-attachments/assets/448930d1-33f7-475a-9a2a-58004c1f3f57">

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll

nihal467 commented:
![image](https://github.com/user-attachments/assets/cead0802-1ee5-46a6-a2f0-8d062f0c4112)

@Jacobjeevan the persist in Chrome brpwser, responsive view of iPhone 14 pro, can you look it

Jacobjeevan committed:
Adjusted padding

- Adjusted padding for result (spacing for inputText in mobile and tab view)
- Removed the print classnames (no longer used as this page isn't for print)

Jacobjeevan commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/376189765-cead0802-1ee5-46a6-a2f0-8d062f0c4112.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MDg3NDcsIm5iZiI6MTcyODkwODQ0NywicGF0aCI6Ii81NzA1NTk5OC8zNzYxODk3NjUtY2VhZDA4MDItMWVlNS00NmE2LWEyZjAtOGQwNjJmMGM0MTEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDEyMjA0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4Zjg5ZDNkNTFjYTMxYmNhYTFlOGE5ODZlZGQ5YmFlODM3NWM4Njg3NjYyODBmMTE0N2E3MzYxMzY3ZmVhYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rwjPE9xYr8LtvViyJVI3bkgIsddB6nWJfO7ZP80FhK8)
> 
> @Jacobjeevan the persist in Chrome browser, responsive view of iPhone 14 pro, can you look it

@nihal467 Got it 👍  Take a look now.

nihal467 commented:
LGTM

Jacobjeevan committed:
Merge branch 'develop' into issues/8583-investigationsView-noScroll
